include/sparrow/buffer.hpp

```diff
@@ -26,12 +26,23 @@ namespace sparrow
         struct buffer_data
         {
             using value_type = T;
+            using reference = T&;
+            using const_reference = const T&;
             using pointer = T*;
             using size_type = std::size_t;
             
             bool empty() const noexcept;
             size_type size() const noexcept;
 
+            reference operator[](size_type);
+            const_reference operator[](size_type) const;
+
+            reference front();
+            const_reference front() const;
+
+            reference back();
+            const_reference back() const;
+
```

Comment:
How about `begin` and `end`? Is this not a range/container?

Reply:
yeah they should be, and I want to refactor this hierarchy to be consistent with the dynamic bitset. We can add it in an incoming PR though, to keep this one quite "small" (it is actually already big)

Reply:
Consider adding TODOs for these so that we dont forget in further passes.